graphql/mutation_resolver.go

```diff
@@ -149,18 +149,29 @@ func (r *mutationResolver) SetAnnotationMetadataLinks(ctx context.Context, taskI
 
 // SaveAdminSettings is the resolver for the saveAdminSettings field.
 func (r *mutationResolver) SaveAdminSettings(ctx context.Context, adminSettings restModel.APIAdminSettings) (*restModel.APIAdminSettings, error) {
-	oldSettings, err := evergreen.GetConfigWithoutSecrets(ctx)
+	oldSettings, err := evergreen.GetConfig(ctx)
 	if err != nil {
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting Evergreen configuration: %s", err.Error()))
 	}
-	_, err = data.SetEvergreenSettings(ctx, &adminSettings, oldSettings, mustHaveUser(ctx), true)
+	newSettings, err := data.SetEvergreenSettings(ctx, &adminSettings, oldSettings, mustHaveUser(ctx), false)
 	if err != nil {
 		return nil, InternalServerError.Send(ctx, fmt.Sprintf("setting admin settings: %s", err.Error()))
 	}
+	if err = newSettings.Validate(); err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("setting admin settings: %s", err.Error()))
+	}
 
-	newSettings, err := evergreen.GetConfig(ctx)
+	oldSettingsRedacted, err := evergreen.GetConfigWithoutSecrets(ctx)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("getting Evergreen configuration: %s", err.Error()))
+	}
+	apiAdminSettings := restModel.NewConfigModel()
```

Comment:
nit: can we add the word "new" in there just for clarity

Reply:
yer